src/parser/core/timePeriods.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * A collection of types and functions that help with the creation and manipulation of time periods.
+ * All time fields are implicitly assumed to be log timestamps i.e. milliseconds since start of logging.
+ */
+
+import { Info } from 'parser/core/metric';
+
+/**
+ * A standard time period with explicitly defined start and end times.
+ */
+export type ClosedTimePeriod = {
+  start: number;
+  end: number;
+};
+
+/**
+ * An 'open' time period whose start and/or end times might not be defined.
+ * Open time periods may be bounded by its enclosing time period during calculations.
+ *
+ * For example, a buff's duration might be an open time period with respect to the encounter,
+ * where the buff might not have a start time if it was applied before the encounter
+ * and might not have an end time if it was still active when the encounter ended.
+ * In this case, the fight's duration is the enclosing time period.
+ */
+export type OpenTimePeriod = Partial<ClosedTimePeriod>;
+
+/** Returns the total duration of the given time period(s).
+ * Overlap will NOT be considered, use union first if non-overlapping duration is required. */
```

Comment:
Isn't overlap actually considered though? If I understand this correctly any overlap will both increase the total.

Reply:
Err yeah this is a weird double negative - I just need to reword this. You're correct, overlap will both increase the total.